src/RandomIterator.cpp

```diff
@@ -0,0 +1,64 @@
+#include "iterlib/RandomIterator.h"
+
+#include <folly/Random.h>
+
+namespace iterlib {
+
+bool RandomIterator::doNext() {
```

Comment:
How do you propose we do that without knowing how many items are there in the inner iterator? Related discussion:

http://stackoverflow.com/questions/24172693/iterative-or-lazy-reservoir-sampling